azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java

```diff
@@ -7,51 +7,82 @@
 import com.microsoft.azure.management.Azure.Authenticated;
-
+import com.microsoft.azure.management.resources.Subscription;
 import org.apache.commons.lang3.StringUtils;
 
 import java.io.IOException;
+import java.util.Optional;
 import java.util.concurrent.ExecutionException;
 
 
-
 public class AzureClientFactory {
+
+    private static final String SUBSCRIPTION_TEMPLATE = "Subscription : %s(%s)";
+    public static final String SUBSCRIPTION_NOT_FOUND = "Subscription %s was not found in current account";
+    public static final String NO_AVAILABLE_SUBSCRIPTION = "No available subscription found in current account";
+
     public static Azure getAzureClient(AuthConfiguration auth, String subscriptionId) throws InvalidConfigurationException, IOException,
         AzureLoginFailureException, InterruptedException, ExecutionException {
-        AzureTokenCredentials azureTokenCredentials = AzureAuthHelper.getAzureTokenCredentials(auth);
+        AzureTokenCredentialsDecorator azureTokenCredentials = AzureAuthHelper.getAzureTokenCredentials(auth);
         if (azureTokenCredentials == null) {
+            AuthType authType;
             final AzureEnvironment environment = AzureEnvironment.AZURE;
             AzureCredential azureCredential;
             try {
+                authType = AuthType.OAUTH;
                 azureCredential = AzureAuthHelper.oAuthLogin(environment);
             } catch (DesktopNotSupportedException e) {
+                authType = AuthType.DEVICE_LOGIN;
                 azureCredential = AzureAuthHelper.deviceLogin(environment);
             }
             AzureAuthHelper.writeAzureCredentials(azureCredential, AzureAuthHelper.getAzureSecretFile());
-            azureTokenCredentials = AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment);
+            azureTokenCredentials = new AzureTokenCredentialsDecorator(authType,
+                    AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment));
         }
 
         if (azureTokenCredentials != null) {
+            // Todo: change to united log
+            Log.info(azureTokenCredentials.getCredentialDescription());
             final Authenticated authenticated = Azure.configure().authenticate(azureTokenCredentials);
             // For cloud shell, use subscription in profile as the default subscription.
             if (StringUtils.isEmpty(subscriptionId) && AzureAuthHelperLegacy.isInCloudShell()) {
                 subscriptionId = AzureAuthHelperLegacy.getSubscriptionOfCloudShell();
             }
-            return StringUtils.isEmpty(subscriptionId) ?
+            final Azure azure = StringUtils.isEmpty(subscriptionId) ?
                     authenticated.withDefaultSubscription() :
-                        authenticated.withSubscription(subscriptionId);
-
+                    authenticated.withSubscription(subscriptionId);
+            checkSubscription(azure, subscriptionId);
```

Comment:
if subscriptionId is empty and there is default subscription id from "azure cli" etc, should use this default id

Reply:
Will check default subscription and show warning if no subscription(default or pom) specified